pkg/externalprovider/provider/base.go

```diff
@@ -0,0 +1,10 @@
+package provider
+
+// ExternalProvider is a provider that is not maintained by the core installer team.
```

Comment:
Perfect example of [Conway's Law](https://en.wikipedia.org/wiki/Conway%27s_law)

I don't think the providers should be grouped based on installer team/non-installer team.

Reply:
The API is designed to be consumable by all providers, but I'm pragmatic about the adoption.